packages/kit/src/core/node/index.js

```diff
@@ -1,44 +1,46 @@
 /**
  * @param {import('http').IncomingMessage} req
- * @returns {Promise<string | Uint8Array>}
+ * @returns {Promise<Uint8Array | string | null>}
  */
 export function getRawBody(req) {
 	return new Promise((fulfil, reject) => {
 		const h = req.headers;
 
 		if (!h['content-type']) {
-			fulfil(null);
-			return;
+			return fulfil(null);
 		}
 
 		req.on('error', reject);
 
 		const length = Number(h['content-length']);
 
-		/** @type {Uint8Array} */
-		let data;
-
-		if (!isNaN(length)) {
-			data = new Uint8Array(length);
+		// https://github.com/lukeed/polka/blob/next/packages/parse/index.js#L17
+		// https://github.com/jshttp/type-is/blob/c1f4388c71c8a01f79934e68f630ca4a15fffcd6/index.js#L81-L95
+		if (isNaN(length) && h['transfer-encoding'] == null) {
+			return fulfil(null);
+		}
 
-			let i = 0;
+		let data = new Uint8Array(length || 0);
 
+		if (length > 0) {
+			let offset = 0;
 			req.on('data', (chunk) => {
-				data.set(chunk, i);
-				i += chunk.length;
-			});
-		} else {
-			// https://github.com/jshttp/type-is/blob/c1f4388c71c8a01f79934e68f630ca4a15fffcd6/index.js#L81-L95
-			if (h['transfer-encoding'] === undefined) {
-				fulfil(null);
-				return;
-			}
+				const new_len = offset + Buffer.byteLength(chunk);
```

Comment:
Using the `chunk.length` was a bug. Incoming `chunks` can be strings, but even if they're still Buffer instances, `length` is not always the same as `byteLength`, which is what `content-length` tracks